pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -141,8 +141,8 @@ public synchronized void removeConsumer(Consumer consumer) throws BrokerServiceE
                 if (log.isDebugEnabled()) {
                     log.debug("[{}] Consumer are left, reading more entries", name);
                 }
-                consumer.getPendingAcks().forEach((pendingMessages, totalMsg) -> {
-                    messagesToReplay.add(pendingMessages);
+                consumer.getPendingAcks().forEach((ledgerId, entryId, batchSize, none) -> {
+                    messagesToReplay.add(PositionImpl.get(ledgerId, entryId));
```

Comment:
just use the `new PositionImpl()` since this instance would hardly be possible to recycle anyway.